module/library/categories/helper.js

```diff
@@ -0,0 +1,147 @@
+      return new Promise(async (resolve, reject) => {
+        
+        try {
+          
+          let queryObject = {};
+
+          if (findQuery != "all") {
+              queryObject = findQuery;
+          }
+
+          let projection = {};
+
+          if (fields != "all") {
+              fields.forEach(element => {
+                  projection[element] = 1;
+              });
+          }
+
+          if (skipFields != "none") {
+              skipFields.forEach(element => {
+                  projection[element] = 0;
+              });
+          }
+
+          let libraryCategoriesData = 
+          await database.models.libraryCategories.find(
+            queryObject, 
+            projection
+          ).lean();
+          
+          return resolve(libraryCategoriesData);
+
+      } catch (error) {
+          return reject({
+              status: error.status || httpStatusCode.internal_server_error.status,
+              message: error.message || httpStatusCode.internal_server_error.message,
+              errorObject: error
+          });
+        }
+      });
+    }
+
+    /**
+      * List of library categories.
+      * @method
+      * @name list
+      * @returns {Object} Library categories lists.
+     */
+
+    static list() {
+        return new Promise(async (resolve, reject) => {
+            try {
+                
+                let libraryCategories = 
+                await this.categoryDocuments(
+                    {
+                        status : messageConstants.common.ACTIVE_STATUS
+                    },
+                    ["externalId","name","icon"]
+                );
+
+                if( !libraryCategories.length > 0 ) {
+                    throw {
+                        status : httpStatusCode.bad_request.status,
+                        message : messageConstants.apiResponses.LIBRARY_CATEGORIES_NOT_FOUND
+                    }
+                }
+
+                let categories = {};
+
+                let libraryCategoriesIcon = libraryCategories.map(category=>{
+                    
+                    categories[category.icon] = {
+                        name : category.name,
+                        type : category.externalId
+                    };
+
+                    return category.icon
+                });
+
+                let downloadableImages = await kendraService.gcpDownloadableImage(
+                    {
+                        filePaths : libraryCategoriesIcon,
+                        bucketName : process.env.GCP_BUCKET_NAME
+                    }
+                );
+
+                if( downloadableImages.status !== httpStatusCode.ok.status) {
+                    throw {
+                        status : httpStatusCode.bad_request.status,
+                        message : messageConstants.apiResponses.URL_COULD_NOT_BE_FOUND
+                    }
+                }
+
+                downloadableImages = downloadableImages.result.map(downloadableImage=>{
```

Comment:
@aman-tunerlabs - The categroy list API response can be cached for faster retrieval as it not going to change often